sn_transfers/src/wallet/authentication.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::wallet::encryption::EncryptedSecretKey;
+use crate::wallet::{Error, Result};
+use chrono::{DateTime, Duration, Utc};
+use secrecy::{ExposeSecret, Secret};
+use std::path::PathBuf;
+
+/// Time (in seconds) before the user has to provide the password again for an encrypted wallet
+const PASSWORD_EXPIRATION_TIME_SECS: i64 = 120;
+
+/// Manager that makes it easier to interact with encrypted wallets
+pub struct AuthenticationManager {
+    /// Password to decrypt the wallet.
+    /// Wrapped in Secret<> so that it doesn't accidentally get exposed
+    password: Option<Secret<String>>,
+    /// Expiry time of the password.
+    /// Has to be provided by the user again after a certain amount of time
+    password_expires_at: Option<DateTime<Utc>>,
+    /// Path to the root directory of the wallet
+    wallet_dir: PathBuf,
+}
```

Comment:
I think the wallet ends up in a file dumped on disk and this struct is a field of wallet. 
So if the `password` is `Some`, it ends up dumped to the disk in clear too. 
Even after expiry, an attacker could read the password by `cat`'ing the wallet file. 
Or did I miss something?

Reply:
Isn't the wallet file only a watch-only-wallet and not a hotwallet? Also, the `Secret<>` wrapper from the [secrecy crate](https://github.com/iqlusioninc/crates/tree/main/secrecy), prevents just that scenario by not offering a serialize impl and the debug impl will automatically redact the inner value.

Reply:
> Isn't the wallet file only a watch-only-wallet and not a hotwallet? 

That I'm not sure.

> Also, the Secret<> wrapper from the [secrecy crate](https://github.com/iqlusioninc/crates/tree/main/secrecy), prevents just that scenario

Okay this is nice! We're safe then!